src/middlewares/authJwt.js

```diff
@@ -0,0 +1,28 @@
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+const verifyToken = (req, res, next) => {
+	const authHeader = req.headers.authorization;
+	const token = authHeader && authHeader.split(' ')[1];
+	if (token == null)
+		return res.status(401).send({
+			status: 'fail',
+			code: 401,
+			message: 'Unauthorized',
+		});
+
+	jwt.verify(token, process.env.ACCESS_TOKEN_SECRET, (err, user) => {
+		if (err)
+			return res.status(401).send({
+				status: 'error',
+				code: 401,
+				message: err,
+			});
+		req.user = user;
```

Comment:
Make sure that the user being decrypted have a **`role`** property otherwise, on **protected routes** the response will always be unauthorized.